src/components/Card.js

```diff
@@ -1,44 +1,70 @@
 export default class Card {
-  constructor({ name, link }, cardSelector, handleCardClick) {
+  constructor({ name, link, _id, isLiked }, cardSelector, handleCardClick, handleDeleteCard, handleLikeCard ) {
+    this.id = _id;
     this._name = name;
     this._link = link;
+    this._isLiked = isLiked;
     this._handleCardClick = handleCardClick;
+    this._handleDeleteCard = handleDeleteCard;
     this._cardSelector = cardSelector;
+    this._handleLikeCard = handleLikeCard;
+
+    this._cardElement = null; // Initialize _cardElement
+
   }
 
-  //EVENT LISTENERS
+
   _setEventListeners() {
-    //card like button
+    // Ensure _cardElement is defined before adding event listeners
+    if (!this._cardElement) return;
+
+    // Card like button
     this._cardElement
       .querySelector(".cards__like-button")
       .addEventListener("click", () => {
-        this._handleLikeIcon();
+        this._handleLikeCard(this);
       });
 
-    //card delete button
+    // Card delete button
     this._cardElement
       .querySelector(".cards__delete-button")
       .addEventListener("click", () => {
-        this._handleDeleteIcon();
+        this._handleDeleteCard(this);
       });
 
-    //open image preview
+    // Open image preview
     this._cardElement
       .querySelector(".cards__image")
       .addEventListener("click", () => {
         this._handleCardClick(this._name, this._link);
       });
   }
 
-  //EVENT HANDLERS
-  _handleLikeIcon() {
-    this._cardElement
-      .querySelector(".cards__like-button")
-      .classList.toggle("cards__like-button_active");
+  updateLikeValue(isLiked) {
+    this._isLiked = isLiked;
+    this._updateLikeButton();
+  }
+
+  getIsLiked() {
+    return this._isLiked;
+  }
+
+  getId() {
+    return this.id;
   }
 
-  _handleDeleteIcon() {
+  _updateLikeButton() {
+    const likeButton = this._cardElement.querySelector(".cards__like-button");
```

Comment:
**Could be improved**

All elements found **every time** you call a method should be defined as **class fields** in the constructor, `_setEventListeners` (or in `getView`) only **1 time** in order not to waste resources on searching them again and again.

`Class fields` (the so-called **class variables** with `this`) are accessible inside any class method. You shouldn’t pass **class fields** into the call of any method or search them again because you can use class fields inside all methods.
There can be `this._inputs`, `this._submitButton`, `this._cardImage`, `this._form`, `this._likeButton`, and so on.

```js
  getView() {
    this._element = this._getTemplate();
    this._likeButton = this._element.querySelector(".card__like-button");
    this._cardImage = this._element.querySelector(".card__image");
    this._trashButton = this._element.querySelector(".card__trash-button");

```